### ISSUE TITLE
Faker and Chance stopped working Since version 4.5

### ISSUE THREAD

**Issue by osher** [json-schema-faker/json-schema-faker#363]
Faker and Chance stopped working Since version 4.5.
Even on the public examples:
 - http://json-schema-faker.js.org/#gist/c0084695b4ca1c4cd015ded1f5c6dc33
 - http://json-schema-faker.js.org/#gist/7cdf200c27eceb6163a79fbc50813fcb

links taken from: https://github.com/json-schema-faker/json-schema-faker#swagger-extensions

2. the result you got
```
Error: Cannot read property 'guid' of null in /properties/...
```
3. the result you expected
It should give a generated example

**Comment by ducin** [comment]
confirmed :(

**Comment by pateketrueke** [comment]
Yeah, probably they are not being bundled anymore...

I suggest you upgrade to latest RC-12 and then register manually both dependencies.

I'll do the same on the website but unfortunately the hosted versions from CDNJS will not longer include third-party dependencies (aka generators).

**Comment by ducin** [comment]
_will not longer include third-party dependencies_
why?

**Comment by pateketrueke** [comment]
> _will not longer include third-party dependencies_
>  why?

Well, this is related with #133 (if we keep bundling everything the maintenance becomes a pain)

I see this should be true for the next major (0.5.x) but not necessarily _required_ for current stable (0.4.x), this was a mistake from my own... I'll fix this asap.

Starting from 0.5.x all third-party that aren't from the core (json-schema-ref-parser, json-path-plus, deref, randexp, tslib) will just bloat our repository, cdnjs, etc.

Also this prevents us from shipping bugs with third-party dependencies, the end-user shall be responsible of picking the right version of, e.g. `faker`or `chance`, etc.

> the hosted versions from CDNJS [are broken]

This is true for 0.4.5 and 0.4.6 because they're already published, so it can be fixed on 0.4.8 but I'm not sure if we're required to remove those broken tags... I guess yes.

**Comment by osher** [comment]
I think that publishing a latest 0.4.x working version will give an initial response.
As long as 0.5 is not released - the default distribution served is 0.4.6 - and that ends with a broken code out of the box, and publishing a working 0.4.x will fix that (unless a user asks for these broken versions specifically)

Removing 0.4.5 and 0.4.6 is the extra mile I believe is to your discretion.
I'm not sure what will be more honorable, to remove them or just to note them in the changelog as broken.

**Comment by pateketrueke** [comment]
Honestly, I don't plan to fall on this again. 🤕 

Just for the sake of this task, building the lib + faker takes 224secs, I know is _nothing_ but also takes unnecessary disk space (~35MB), or bandwith when publishing/downloading, etc.

I plan to fix the web version to always have faker/chance available on `window` just for demo purposes... I hope this happens soon!

From nodejs side, it's encouraged to migrate to `0.5.x` since all new features are implemented there.

**Comment by pateketrueke** [comment]
I finally published `0.4.7` which only resolves the problem for the hosted versions of `json-schema-faker`, so bower and cdnjs now should have complete versions of the distributed files.

Sadly I had to remove those tags since they are used for indexing on bower registry. 😢

**Comment by osher** [comment]
+1

This should do until 0.5.x will be published without tags.
